fail2ban/server/filter.py

```diff
@@ -468,7 +468,7 @@ def performSvc(self, force=False):
 
 	def addAttempt(self, ip, *matches):
 		"""Generate a failed attempt for ip"""
-		if not isinstance(ip, IPAddr):
+		if not isinstance(ip, (IPAddr, list, tuple)):
```

Comment:
Does it really bother somewhere? If so then we'd need to fix that in `IPAddr`, because normally `IPAddr` can safe handle raw values (not IPs, inclusive tuple or lists) without any problem. But has additional facility to check whether it's IP, etc. 
So we'd always prefer `IPAddr` as identifier/IP in internal filters/actions functionality if possible.
```python
>>> from fail2ban.server.ipdns import IPAddr
>>> id = ('A','B')
>>> ip = IPAddr(id); ip
('A', 'B')
>>> ip == id
True
>>> ip.family == IPAddr.CIDR_RAW
True
>>> ip.raw
('A', 'B')
```